lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -2235,6 +2265,56 @@ constructTransaction ctx genChange knownPools getPoolStatus (ApiT wid) body = do
                     , wallet
                     , selectionStrategy = SelectionStrategyOptimal
                     }
+        txCtx' <-
+            if isJust mintingBurning then do
+                (policyXPub, _) <-
+                    liftHandler $ W.readPolicyPublicKey @_ @s @k @n wrk wid
+                let isMinting (ApiMintBurnData _ _ (ApiMint _)) = True
+                    isMinting _ = False
+                let getMinting = \case
+                        ApiMintBurnData
+                            (ApiT scriptT)
+                            (ApiT tName)
+                            (ApiMint (ApiMintData _ (Quantity amt))) ->
+                            toTokenMapAndScript @k
+                                scriptT
+                                (Map.singleton (Cosigner 0) policyXPub)
+                                tName
+                                amt
+                        _ -> error "getMinting should not be used that way"
```

Comment:
@paweljakubas is this `error` meant to represent a **_programming error_**, or it is possible for the user to trigger it from a faulty API call?

In an case, suggest changing the text to:

```suggestion
                        _ -> error "getMinting should not be used in this way"
```

Reply:
it is just impossible  type error in a sense that getMinting is used for values filtered to be minting. Corrected the wording.